light-client/src/supervisor.rs

```diff
@@ -248,11 +248,9 @@ impl Supervisor {
                 }
             }
             // Verification failed
-            Err(_err) => {
-                // TODO: Log/record error
-
+            Err(err) => {
                 // Swap primary, and continue with new primary, if there is any witness left.
-                self.peers.replace_faulty_primary()?;
+                self.peers.replace_faulty_primary(Some(err))?;
```

Comment:
When putting together the light-node, I also stumbled upon this: I was seeing "no witness left" and it wasn't really informing what went wrong. 
@romac does anything speak against this approach? The todo is indicating you already had sth in mind for this?